File: Data Structures & Algorithms/find-lucky-integer-in-an-array/submission-0.py
class Solution:
    def findLucky(self, arr: List[int]) -> int:
        hashmap = {}
        res = -1

        for num in arr:
            if num not in hashmap:
                hashmap[num] = 1
            else:
                hashmap[num] += 1
        
        for key, val in hashmap.items():
            if key == val:
                res = max(res, key)
        
        return res

File: Data Structures & Algorithms/number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold/submission-0.java
class Solution {
    public int numOfSubarrays(int[] arr, int k, int threshold) {
        int l = 0;
        int r = k - 1;
        int res = 0;
        int sum = 0;

        while (r < arr.length) {
            System.out.println(l + " " + r);
            for(int i = l; i <= r; i++) {
                sum += arr[i];
            }
            int avg = sum / k;
            System.out.println(avg);
            if (avg >= threshold) {
                res += 1;
            }
            
            l++;
            r++;
            sum = 0;
        }
        return res;
    }
}